pyproj/warn.py

```diff
@@ -0,0 +1,16 @@
+class PyProjWarning(UserWarning):
+    """
+    This is a warning for pyproj
+    """
+
+
+class PyProjDeprecationWarning(PyProjWarning):
```

Comment:
Is there a reason you went with a custom warning instead of using the default DeprecationWarning? 

Also, I think it is better to at least subclass from DeprecationWarning, so people can catch just DeprecationWarning.

Reply:
In principle, I agree with you that it could subclass from DeprecationWarning.

However, by default the DeprecationWarning is filtered out:
```python
>>> import warnings
>>> def deprecated():
...     warnings.warn("deprecated", DeprecationWarning)
... 
>>> deprecated()
>>> def user_deprecated():
...     warnings.warn("user_deprecated", UserWarning)
... 
>>> user_deprecated()
__main__:2: UserWarning: user_deprecated

```
And I find that not very useful if you want to give someone heads up that they need to change something.

Also, with a custom instance of a warning, the user can filter based on that specific warning instance.

```python
with warnings.catch_warnings():
    warnings.simplefilter("ignore", ProjDeprecationWarning)
    ...
```

What are your thoughts?